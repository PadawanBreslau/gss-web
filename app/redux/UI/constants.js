/*
 *
 * Messages constants
 *
 */

export const SHOW_MESSAGE = 'app/Messages/SHOW_MESSAGE';
export const HIDE_MESSAGE = 'app/Messages/HIDE_MESSAGE';

export const SHOW_LOADING = 'app/Messages/SHOW_LOADING';
export const HIDE_LOADING = 'app/Messages/HIDE_LOADING';
