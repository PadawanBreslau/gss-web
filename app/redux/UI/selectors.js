import { createSelector } from 'reselect';

/**
 * Direct selector to the messages state domain
 */
const selectUiDomain = (state) => state.ui;

/**
 * Other specific selectors
 */

/**
 * Default selector used by Messages
 */

const makeSelectUi = () =>
  createSelector(
    selectUiDomain,
    (substate) => substate,
  );

export default makeSelectUi;
export { selectUiDomain };
