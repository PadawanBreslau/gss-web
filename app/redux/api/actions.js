import {
  RELOAD_PAGEDATA_REQUEST,
  INITIALIZE_PAGEDATA_REQUEST,
  LOAD_PAGEDATA_SUCCESS,
  LOAD_PAGEDATA_ERROR,
  SUBMIT_PAGEDATA_REQUEST,
  SUBMIT_PAGEDATA_SUCCESS,
  SUBMIT_PAGEDATA_ERROR,
} from './constants';

export function reloadStore(storeName) {
  return {
    type: `${RELOAD_PAGEDATA_REQUEST}/${storeName}`,
  };
}

export default function generateActions(storeName) {
  return {
    initializePageData: (endpoint, callback) => ({
      type: `${INITIALIZE_PAGEDATA_REQUEST}/${storeName}`,
      endpoint,
      callback,
    }),
    reloadPageData: (endpoint, callback) => ({
      type: `${RELOAD_PAGEDATA_REQUEST}/${storeName}`,
      endpoint,
      callback,
    }),
    loadPageDataSuccess: (payload) => ({
      type: `${LOAD_PAGEDATA_SUCCESS}/${storeName}`,
      payload,
    }),
    loadPageDataError: (error) => ({
      type: `${LOAD_PAGEDATA_ERROR}/${storeName}`,
      error,
    }),

    submitPageData: (endpoint, method, payload, successCallback, id) => ({
      type: `${SUBMIT_PAGEDATA_REQUEST}/${storeName}`,
      endpoint,
      method,
      payload: {
        data: {
          attributes: payload,
        },
      },
      successCallback,
      id,
    }),
    submitPageDataSuccess: (payload, id) => ({
      type: `${SUBMIT_PAGEDATA_SUCCESS}/${storeName}`,
      payload,
      id,
    }),
    submitPageDataError: (error, id) => ({
      type: `${SUBMIT_PAGEDATA_ERROR}/${storeName}`,
      error,
      id,
    }),
  };
}
