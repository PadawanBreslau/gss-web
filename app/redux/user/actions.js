import {
  LOGOUT_REQUEST,
  LOGOUT_SUCCESS,
  LOGOUT_ERROR,
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
  SIGN_UP_REQUEST,
  SIGN_UP_SUCCESS,
  SIGN_UP_ERROR,
  PASSWORD_RESET_REQUEST,
  PASSWORD_RESET_REQUEST_SUCCESS,
  PASSWORD_RESET_REQUEST_ERROR,
  SET_PASSWORD_REQUEST,
  SET_PASSWORD_SUCCESS,
  SET_PASSWORD_ERROR,
  GOOGLE_LOGIN_REQUEST,
} from './constants';

export function LogOut(endpoint, successCallback) {
  return {
    type: LOGOUT_REQUEST,
    endpoint,
    successCallback,
  };
}
export function LogOutSuccess() {
  return {
    type: LOGOUT_SUCCESS,
  };
}
export function LogOutError(error) {
  return {
    type: LOGOUT_ERROR,
    error,
  };
}
export function LogIn(endpoint, payload, successCallback) {
  return {
    type: LOGIN_REQUEST,
    endpoint,
    payload: {
      data: {
        attributes: payload,
      },
    },
    successCallback,
  };
}
export function LogInSuccess(data) {
  return {
    type: LOGIN_SUCCESS,
    data,
  };
}
export function LogInError(error) {
  return {
    type: LOGIN_ERROR,
    error,
  };
}
export function SignUp(endpoint, payload, successCallback) {
  return {
    type: SIGN_UP_REQUEST,
    endpoint,
    payload: {
      data: {
        attributes: payload,
      },
    },
    successCallback,
  };
}
export function SignUpSuccess(data) {
  return {
    type: SIGN_UP_SUCCESS,
    data,
  };
}
export function SignUpError(error) {
  return {
    type: SIGN_UP_ERROR,
    error,
  };
}
export function PasswordReset(endpoint, payload) {
  return {
    type: PASSWORD_RESET_REQUEST,
    endpoint,
    payload: {
      data: {
        attributes: payload,
      },
    },
  };
}
export function PasswordResetRequestSuccess(data) {
  return {
    type: PASSWORD_RESET_REQUEST_SUCCESS,
    data,
  };
}
export function PasswordResetRequestError(error) {
  return {
    type: PASSWORD_RESET_REQUEST_ERROR,
    error,
  };
}
export function SetPassword(endpoint, payload) {
  return {
    type: SET_PASSWORD_REQUEST,
    endpoint,
    payload: {
      data: {
        attributes: payload,
      },
    },
  };
}
export function SetPasswordSuccess(data) {
  return {
    type: SET_PASSWORD_SUCCESS,
    data,
  };
}
export function SetPasswordError(error) {
  return {
    type: SET_PASSWORD_ERROR,
    error,
  };
}
export function GoogleLogIn(googleUserObject, successCallback) {
  return {
    type: GOOGLE_LOGIN_REQUEST,
    googleUserObject,
    successCallback,
  };
}
