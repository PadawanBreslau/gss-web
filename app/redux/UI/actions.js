import { SHOW_MESSAGE, HIDE_MESSAGE, SHOW_LOADING, HIDE_LOADING } from './constants';

export function showLoading() {
  return {
    type: SHOW_LOADING,
  };
}
export function hideLoading() {
  return {
    type: HIDE_LOADING,
  };
}
export function showError(title, text) {
  return {
    type: SHOW_MESSAGE,
    data: {
      type: 'error',
      title,
      text,
    },
  };
}
export function showUiError(error) {
  return {
    type: SHOW_MESSAGE,
    data: {
      type: 'error',
      error,
    },
  };
}
export function showUiSuccess(text) {
  return {
    type: SHOW_MESSAGE,
    data: {
      type: 'success',
      title: 'Success',
      text,
    },
  };
}
export function hideUiMessage() {
  return {
    type: HIDE_MESSAGE,
  };
}
