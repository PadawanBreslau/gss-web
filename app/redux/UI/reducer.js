/* eslint-disable no-param-reassign */
/*
 *
 * Messages reducer
 *
 */

import produce from 'immer';
import { SHOW_MESSAGE, HIDE_MESSAGE, SHOW_LOADING, HIDE_LOADING } from './constants';

const initialState = {
  message: {
    isActive: false,
    title: null,
    text: null,
    type: null,
  },
  loading: false,
  error: null,
};

const uiReducer = (state = initialState, action) =>
  // eslint-disable-next-line consistent-return
  produce(state, (draft) => {
    switch (action.type) {
      case SHOW_LOADING:
        draft.loading = true;
        draft.message = {
          title: null,
          text: null,
          type: null,
          isActive: false,
        };
        break;
      case HIDE_LOADING:
        draft.loading = false;
        break;
      case SHOW_MESSAGE:
        draft.loading = false;
        draft.message = {
          ...action.data,
          isActive: true,
        };
        break;
      case HIDE_MESSAGE:
        draft.message = {
          title: null,
          text: null,
          type: null,
          isActive: false,
        };
        break;
      default:
        break;
    }
  });

export default uiReducer;
