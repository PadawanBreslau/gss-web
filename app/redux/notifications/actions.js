import { REGISTER_PUSH } from './constants';

export function registerPushMessages() {
  return {
    type: REGISTER_PUSH,
  };
}
