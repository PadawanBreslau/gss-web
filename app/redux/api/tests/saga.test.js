import { put, call, takeLatest } from 'redux-saga/effects';
import { showUiError } from 'redux/UI/actions';
import {
  RELOAD_PAGEDATA_REQUEST,
  INITIALIZE_PAGEDATA_REQUEST,
  SUBMIT_PAGEDATA_REQUEST,
} from '../constants';
import generateActions from '../actions';
import generateSaga, { fetchApiDataSaga, sendApiDataSaga } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('API: fetchApiDataSaga', () => {
  let fetchRequestGenerator;
  const exampleResponse = {
    data: {
      attributes: { some: 'attribute' },
    },
  };
  const { loadPageDataSuccess, loadPageDataError } = generateActions('testStore');
  beforeEach(() => {
    fetchRequestGenerator = fetchApiDataSaga('testStore', {
      endpoint: '/test',
    });
    const putLoadingDescriptor = fetchRequestGenerator.next().value;
    expect(putLoadingDescriptor).toMatchSnapshot();
    const callDescriptor = fetchRequestGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });
  it('should dispatch the loadPageDataSuccess action if it requests the data successfully', () => {
    const putDescriptor = fetchRequestGenerator.next(exampleResponse).value;
    expect(putDescriptor).toEqual(put(loadPageDataSuccess(exampleResponse)));
  });
  it('should initialize form with the fetched data', () => {
    const initializeFormCallback = (data) => ({
      type: 'someAction',
      data,
    });
    const callbackRequestGenerator = fetchApiDataSaga('testStore', {
      endpoint: '/test',
      method: 'post',
      payload: {},
      callback: initializeFormCallback,
    });
    const putLoadingDescriptor = callbackRequestGenerator.next().value;
    expect(putLoadingDescriptor).toMatchSnapshot();
    const callDescriptor = callbackRequestGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
    const putDescriptor = callbackRequestGenerator.next(exampleResponse).value;
    expect(putDescriptor).toEqual(put(loadPageDataSuccess(exampleResponse)));
    const putLoaderDescriptor = callbackRequestGenerator.next().value;
    expect(putLoaderDescriptor).toMatchSnapshot();
    const putCallbackDescriptor = callbackRequestGenerator.next(exampleResponse).value;
    expect(putCallbackDescriptor).toEqual(
      call(initializeFormCallback, {
        data: { attributes: { some: 'attribute' } },
      }),
    );
  });
  it('should call the loadPageDataError action if the response errors', () => {
    const error = {
      title: 'some api error',
    };
    const putShowErrorDescriptor = fetchRequestGenerator.throw(error).value;
    expect(putShowErrorDescriptor).toEqual(put(showUiError(error)));
    const putErrorDescriptor = fetchRequestGenerator.next().value;
    expect(putErrorDescriptor).toEqual(put(loadPageDataError(error)));
  });
});

describe('API: sendApiDataSaga', () => {
  let sendRequestGenerator;
  const exampleResponse = {};
  const { submitPageDataSuccess, submitPageDataError } = generateActions('testStore');
  beforeEach(() => {
    sendRequestGenerator = sendApiDataSaga('testStore', {
      endpoint: '/test',
      method: 'post',
      payload: {},
    });
    const putLoadingDescriptor = sendRequestGenerator.next().value;
    expect(putLoadingDescriptor).toMatchSnapshot();
    const callDescriptor = sendRequestGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });
  it('should dispatch the submitPageDataSuccess action if it requests the data successfully', () => {
    const putDescriptor = sendRequestGenerator.next(exampleResponse).value;
    expect(putDescriptor).toEqual(put(submitPageDataSuccess(exampleResponse)));
  });
  it('should do the callback after success', () => {
    const successCallback = (data) => ({
      type: 'someAction',
      data,
    });
    sendRequestGenerator = sendApiDataSaga('testStore', {
      endpoint: '/test',
      method: 'post',
      payload: {},
      successCallback,
    });
    const putLoadingDescriptor = sendRequestGenerator.next().value;
    expect(putLoadingDescriptor).toMatchSnapshot();
    const callDescriptor = sendRequestGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
    const putDescriptor = sendRequestGenerator.next(exampleResponse).value;
    expect(putDescriptor).toEqual(put(submitPageDataSuccess(exampleResponse)));
    const putCallbackDescriptor = sendRequestGenerator.next(exampleResponse).value;
    expect(putCallbackDescriptor).toEqual(put(successCallback));
  });

  it('should call the submitPageDataError action if the response errors', () => {
    const error = {
      title: 'some api error',
    };
    const putShowErrorDescriptor = sendRequestGenerator.throw(error).value;
    expect(putShowErrorDescriptor).toEqual(put(showUiError(error)));
    const putErrorDescriptor = sendRequestGenerator.next().value;
    expect(putErrorDescriptor).toEqual(put(submitPageDataError(error)));
  });
});

describe('API: defaultSaga watcher', () => {
  const saga = generateSaga('testStore');
  const watcher = saga();
  it('should start task to watch for RELOAD_PAGEDATA_REQUEST/testStore action', () => {
    const takeLatestDescriptor = watcher.next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(`${RELOAD_PAGEDATA_REQUEST}/testStore`, fetchApiDataSaga, 'testStore'),
    );
  });
  it('should start task to watch for INITIALIZE_PAGEDATA_REQUEST/testStore action', () => {
    const takeLatestDescriptor = watcher.next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(`${INITIALIZE_PAGEDATA_REQUEST}/testStore`, fetchApiDataSaga, 'testStore'),
    );
  });
  it('should start task to watch for SUBMIT_PAGEDATA_REQUEST/testStore action', () => {
    const takeLatestDescriptor = watcher.next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(`${SUBMIT_PAGEDATA_REQUEST}/testStore`, sendApiDataSaga, 'testStore'),
    );
  });
});
