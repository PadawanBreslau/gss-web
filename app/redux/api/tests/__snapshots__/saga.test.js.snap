// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API: fetchApiDataSaga should call the loadPageDataError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`API: fetchApiDataSaga should call the loadPageDataError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/test",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`API: fetchApiDataSaga should dispatch the loadPageDataSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`API: fetchApiDataSaga should dispatch the loadPageDataSuccess action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/test",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`API: fetchApiDataSaga should initialize form with the fetched data 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`API: fetchApiDataSaga should initialize form with the fetched data 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/test",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`API: fetchApiDataSaga should initialize form with the fetched data 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`API: fetchApiDataSaga should initialize form with the fetched data 4`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/test",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`API: fetchApiDataSaga should initialize form with the fetched data 5`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/Messages/HIDE_LOADING",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`API: sendApiDataSaga should call the submitPageDataError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`API: sendApiDataSaga should call the submitPageDataError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/test",
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`API: sendApiDataSaga should dispatch the submitPageDataSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`API: sendApiDataSaga should dispatch the submitPageDataSuccess action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/test",
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`API: sendApiDataSaga should do the callback after success 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`API: sendApiDataSaga should do the callback after success 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/test",
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`API: sendApiDataSaga should do the callback after success 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`API: sendApiDataSaga should do the callback after success 4`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/test",
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
